sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Storage/BlobsCheckpointStore.Diagnostics.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Messaging.EventHubs.Processor.Diagnostics;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    internal sealed partial class BlobsCheckpointStore
+    {
+        /// <summary>
+        ///   The instance of <see cref="BlobEventStoreEventSource" /> which can be mocked for testing.
+        /// </summary>
+        ///
+        internal BlobEventStoreEventSource Logger { get; set; } = BlobEventStoreEventSource.Log;
+
+        /// <summary>
+        ///   Indicates that an attempt to retrieve a list of ownership has completed.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace the ownership are associated with.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub the ownership are associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group the ownership are associated with.</param>
+        /// <param name="ownershipCount">The amount of ownership received from the storage service.</param>
+        ///
+        partial void ListOwnershipComplete(string fullyQualifiedNamespace, string eventHubName, string consumerGroup, int ownershipCount) =>
```

Comment:
These methods are to isolate the logging code and the BlobsCheckpointStore to be included in project that doesn't have the BlobEventStoreEventSource